### PR TITLE
Activation unit

### PR DESCRIPTION
**Pull request by gamzeisl**
This PR cleans and extends the merge request [!10](https://iis-git.ee.ethz.ch/ita/ita/-/merge_requests/10) on GitLab.

Changes:
- Implementation of integer-only GELU and RELU activation unit for ITA. 
- Feedforward and linear layers are added to the FSM.
- FFN requant params are reserved in the register file.

### CODE REVIEW

File: PyITA/ITA.py [hunk 1]
```diff
@@ -806,14 +971,186 @@ def export_numpy(self):
...
+
+
+def round_to_u8(x: f32) -> u8:
+    x_rounded_clipped: f32 = round_and_clip(x, 8)
+    return x_rounded_clipped.astype(u8)
+
+
+def round_to_i16(x: f32) -> i16:
+    x_rounded_clipped: f32 = round_and_clip(x, 16)
+    return x_rounded_clipped.astype(i16)
```
Comment (Xeratec) [inline PyITA/ITA.py:1001]: A bit of a nitpick but this should be in `util.py`

File: PyITA/ITA.py [hunk 2]
```diff
@@ -806,14 +971,186 @@ def export_numpy(self):
...
+    q_out: i32 = d_sq + q_c32
+    return q_out.astype(i32)
+
+
+def i_poly_wrapper(q: i8, S: f32, a: f32, b: f32, c: f32) -> Tuple[i32, f32]:
+    q_b: i16 = round_to_i16(b / S)
+    q_c: i16 = round_to_i16(c / (a * S**2))
+    S_out: f32 = a * S**2
+    q_out: i32 = i_poly(q, q_b, q_c)
+    return q_out, S_out
```
Comment (Xeratec) [inline PyITA/ITA.py:1095]: This I would place in a new file called `gely.py`

File: PyITA/ITA.py [hunk 3]
```diff
@@ -806,14 +971,186 @@ def export_numpy(self):
...
+    x_q = np.clip(activations, clip_lo, clip_hi)
+    x_q = x_q / S
+    x_q = np.array(list(map(round, x_q)))
+    return x_q, S
+
+
+def almost_symmetric_dequantize(quantized_activations: np.ndarray, clip_lo: f32, n_bits: int = 8) -> np.ndarray:
+    S, _ = get_almost_symmetric_scaling_factor(clip_lo, n_bits)
+    activations = quantized_activations * S
+    return activations
```
Comment (Xeratec) [inline PyITA/ITA.py:1139]: And this again into `util.py`

File: PyITA/test_ITA.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
+import pytest
```
Comment (Xeratec) [inline PyITA/test_ITA.py:1]: A short description of what this file does would be very nice. I assume it does test the newly added feature for GeLU. In this case, I would change the name as it is not a full test for the ITA golden model but rather a subset.

File: .gitlab-ci.yml
[the remark this answers is not in the document]
Comment (gamzeisl) [reply .gitlab-ci.yml:1]: Identity is actually covered by default as gelu is only active during feedforward (f1). I added tests for relu

**Pull request merged**